imagepipeline/src/main/java/com/facebook/imagepipeline/producers/LocalFetchProducer.java

```diff
@@ -105,20 +105,34 @@ protected EncodedImage getByteBufferBackedEncodedImage(
     }
   }
 
-  protected EncodedImage getEncodedImage(
+    private boolean isMemoryEnough() {
+      Runtime runTime = Runtime.getRuntime();
+      long javaMax = runTime.maxMemory();
+      long javaUsed = runTime.totalMemory() - runTime.freeMemory();
+      long javaFree = Math.min(javaMax - javaUsed, 8 * ByteConstants.MB);
+      return (javaMax >= 64 * javaFree);
```

Comment:
@duchuyctlk   **javaMax** contain more free java free memory in the future. **javaFree**  is not constant.
 ` (javaMax >= 64 * javaFree)`  means that java heap has enough memory.